tests/lib.rs

```diff
@@ -51,283 +51,57 @@
 )]
 
 use bytes::Bytes;
-use self_encryption::{encrypt, ChunkInfo, DataMap, Result, MAX_CHUNK_SIZE};
+use self_encryption::{encrypt, ChunkInfo, DataMap, Result};
 use xor_name::XorName;
 
 #[tokio::test]
 async fn cross_platform_check() -> Result<()> {
-    let content_size: usize = 20 * *MAX_CHUNK_SIZE + 100;
+    // Use a fixed size that will give us exactly 3 chunks
+    let content_size: usize = 3 * 1024; // Small enough to not trigger shrinking
     let mut content = vec![0u8; content_size];
-    for (i, c) in content.iter_mut().enumerate().take(content_size) {
-        *c = (i % 17) as u8;
-    }
-
-    let (data_map, _) = encrypt(Bytes::from(content))?;
 
-    // (NB: this hard-coded ref needs update if algorithm changes)
-    let ref_data_map = vec![
-        ChunkInfo {
-            src_hash: XorName([
-                219, 177, 84, 234, 189, 172, 82, 64, 169, 100, 5, 56, 3, 43, 142, 126, 51, 235,
-                194, 243, 30, 130, 132, 197, 137, 36, 170, 62, 46, 44, 176, 201,
-            ]),
-            dst_hash: XorName([
-                248, 155, 46, 153, 173, 52, 226, 212, 133, 172, 107, 200, 72, 150, 41, 50, 116, 77,
-                85, 92, 67, 168, 25, 56, 93, 61, 209, 194, 65, 172, 227, 130,
```

Comment:
Did the algo change? I feel like this kind of test was very useful as they are very strict

Reply:
Yes, as we now shrink data_maps and they all have length==3 this test became invalid. I think we should re-introduce such a test again when all this work is in place as it's very handy. In tis case I Could not differentiate the chunks from the file and data_map though. 

What we did though was create a backend matrix that compares all encrypt and decrypt functions in a matrix to check we have not broken any existing algorithms. It's not as robust as a test like this one though and I agree we should introduce another hard coded one like this soon

Reply:
Yes, the issue is we now shrink the data maps and don't return any > Len == 3 so the test became useless at this time. We should look to create another such test for backwards compatibility again when all these changes are in place.